lib/web/mage/sticky.js

```diff
@@ -41,8 +44,34 @@ define([
 
             if (!isStatic && this.element.is(':visible')) {
                 offset = $(document).scrollTop() - this.parentOffset;
+
+                if (typeof this.options.spacingTop === 'function') {
+                    offset += this.options.spacingTop();
+                } else {
+                    offset += this.options.spacingTop;
+                }
+
                 offset = Math.max(0, Math.min(offset, this.maxOffset));
-                this.element.css('top', offset);
+
+                if (offset &&
+                    this.options.offsetTop &&
+                    !this.element.hasClass(this.options.stickyClass)) {
+
+                    var offsetTop = 0;
+                    if (typeof this.options.offsetTop === 'function') {
+                        offsetTop = this.options.offsetTop();
+                    } else {
+                        offsetTop = this.options.offsetTop;
+                    }
```

Comment:
It's okay to send `this` from an object method to a locally declared function. It's a way to do real privacy in JavaScript. I see it in lots of codebases! But the way that you implemented it, as an underscore-prefixed method, is also just fine.